mettle/api/api.go

```diff
@@ -91,19 +95,59 @@ type server struct {
 	mutex        sync.Mutex
 }
 
-// ServeExecutions serves a list of currently executing jobs over HTTP.
-func (s *server) ServeExecutions(w http.ResponseWriter, r *http.Request) {
-	resp := map[string]string{}
-	m := jsonpb.Marshaler{Indent: "  "}
+// ServeExecutions serves a list of currently executing jobs over GRPC.
+func (s *server) ServeExecutions(ctx context.Context, req *bpb.ServeExecutionsRequest) (*bpb.ServeExecutionsResponse, error) {
+	log.Notice("Received request for inflight executions")
 	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	executions := []*bpb.Job{}
 	for k, v := range s.jobs {
-		s, _ := m.MarshalToString(v.Current)
-		resp[k] = s
+		current, _ := proto.Marshal(v.Current)
+		job := &bpb.Job{
+			ID:        k,
+			Current:   current,
+			SentFirst: v.SentFirst,
+			Done:      v.Done,
+		}
+		executions = append(executions, job)
 	}
-	s.mutex.Unlock()
-	e := json.NewEncoder(w)
-	e.SetIndent("", "  ")
-	e.Encode(resp)
+	res := &bpb.ServeExecutionsResponse{
+		Jobs: executions,
+	}
+	log.Notice("Serving %d inflight executions", len(executions))
+	return res, nil
+}
+
+// getExecutions requests a list of currently executing jobs over grpc and updates the server.
```

Comment:
comment is slightly inaccurate - it doesn't update the server, that's (correctly) done outside this function.